x/epochs/types/identifier.go

```diff
@@ -0,0 +1,43 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	// DayEpochID defines the identifier for a daily epoch.
+	DayEpochID = "day"
+	// HourEpochID defines the identifier for an hourly epoch.
+	HourEpochID = "hour"
+	// MinuteEpochID defines the identifier for an epoch that is a minute long.
+	MinuteEpochID = "minute"
+	// WeekEpochID defines the identifier for a weekly epoch.
+	WeekEpochID = "week"
+)
+
+// ValidateEpochIdentifierInterface accepts an interface and validates it as an epoch
+// identifier. It is not used directly by this module; rather it is created for other
+// modules to validate their params.
+func ValidateEpochIdentifierInterface(i interface{}) error {
+	v, ok := i.(string)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+	if err := ValidateEpochIdentifierString(v); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ValidateEpochIdentifierString accepts a string and validates it as an epoch identifier.
+// It is not used directly by this module; rather it is created for other modules to validate
+// their params.
+func ValidateEpochIdentifierString(s string) error {
```

Comment:
The current ValidateEpochIdentifierString function only checks that the string is not null, it does not verify that the string is one of the defined constants. An additional check can be added to ensure that the string is a valid epoch identifier

Reply:
The epoch identifier can be any arbitrary string. The constants that we have defined only refer to the default epochs. For example, it is possible to set up an identifier called "half-hour" with duration of 30 minutes (or even 20 minutes for that matter; the name and the duration aren't validated).